cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowProgramRunner.java

```diff
@@ -68,15 +103,39 @@ public ProgramController run(Program program, ProgramOptions options) {
 
     WorkflowSpecification workflowSpec = appSpec.getWorkflows().get(program.getName());
     Preconditions.checkNotNull(workflowSpec, "Missing WorkflowSpecification for %s", program.getName());
-    
-    WorkflowDriver driver = new WorkflowDriver(program, options, hostname, workflowSpec, programRunnerFactory);
 
     // Controller needs to be created before starting the driver so that the state change of the driver
     // service can be fully captured by the controller.
     RunId runId = RunIds.fromString(options.getArguments().getOption(ProgramOptionConstants.RUN_ID));
+
+    AdapterDefinition adapterSpec = getAdapterSpecification(options.getArguments());
```

Comment:
Doesn't seem to be used. Please remove. Also remove the ``getAdapterSpecification`` method if this is unused.